datumaro/components/errors.py

```diff
@@ -197,21 +197,66 @@ class DatasetImportError(DatumaroError):
     pass
 
 
+class InvalidAnnotationError(DatasetImportError):
+    """
+    A basic dataset parsing error. Should include the problem description in
+    the message.
+    """
+
+
+@define(auto_exc=False)
+class InvalidFieldError(InvalidAnnotationError):
+    name: str
+
+    def __str__(self) -> str:
+        return f"Invalid anotation field {self.name} value"
+
+
+@define(auto_exc=False)
+class InvalidFieldTypeError(InvalidFieldError):
+    actual: str
+    expected: Tuple[str]
+
+    def __str__(self) -> str:
+        if len(self.expected) == 1:
+            expected = self.expected[0]
+        else:
+            expected = "one of " + ", ".join(self.expected)
+        return f"Invalid anotation field {self.name} type {self.actual}. Expected {expected}"
+
+
+@define(auto_exc=False)
+class MissingFieldError(InvalidAnnotationError):
+    name: str
+
+    def __str__(self) -> str:
+        return f"Missing anotation field {self.name}"
+
+
+@define(auto_exc=False)
+class InvalidLabelError(InvalidAnnotationError):
```

Comment:
I don't think this needs to be a separate exception:

* I don't see a use case for handling it differently from `InvalidAnnotationError`;
* The default message is too generic to be useful (it doesn't explain why the value is invalid).
* Its semantics overlap with other exceptions (e.g. if a field represents a label, and it has the wrong type, is that an invalid label or an invalid field type?).

It's also only used once. I think that use can just be replaced with an `InvalidAnnotationError` with a custom message.

Reply:
> I don't see a use case for handling it differently from InvalidAnnotationError;

Basically, the idea is to group similar errors and display collective error info in the UI. In general, we'd like to have specialized error types instead of a single generic one.

> The default message is too generic to be useful (it doesn't explain why the value is invalid).
> Its semantics overlap with other exceptions (e.g. if a field represents a label, and it has the wrong type, is that an invalid label or an invalid field type?).

It represents exactly the invalid label value, of the right type. In most cases I expect it to indicate a value, that is not declared in the label list of the dataset. Invalid type should be checked before. Maybe, it's better to rename it to "UndeclaredLabel"?

> It's also only used once. I think that use can just be replaced with an InvalidAnnotationError with a custom message.

I expect it to be applicable to almost all formats.

Reply:
> Basically, the idea is to group similar errors and display collective error info in the UI. In general, we'd like to have specialized error types instead of a single generic one.

Okay, that's fair.

> It represents exactly the invalid label value, of the right type. In most cases I expect it to indicate a value, that is not declared in the label list of the dataset.

The problem is that just "invalid" doesn't really say much.

> Maybe, it's better to rename it to "UndeclaredLabel"?

Okay, that's a more specific error, so I think there's a better case for it.